javascript/javascript3/week3/shoppingcartusingclasses.js

```diff
@@ -0,0 +1,106 @@
+const url = 'https://jsonplaceholder.typicode.com/users/1'
+
+class Product {
+    constructor(name, price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    renderInfo() {
+      const myInfo = document.createElement('div')
+
+      this.products.forEach((items) => {
+        const p = document.createElement('p')
+        p.textContent = this.name
+        p.textContent = this.price
+        myInfo.appendChild(p)
+      })
+        return myInfo
+    }
+
+    convertToCurrency(currency) {
+      return fetch('http://data.fixer.io/api/latest?access_key=25e524a669334b226a450e77c8dd3ef8')
+        .then(res => res.json())
+        .then(result => {
+          const rate = result.rates[currency]
+          return rate * this.price
+        })
+    }
+}
+
+class ShoppingCart {
+    constructor() {
+        this.products = [];
+    }
+
+    addProduct(product) {
+      this.products.push(product)
+    }
+
+    removeProduct(product) {
+      const newProduct = this.products.filter((existingProduct) => {
+          return existingProduct.name !== product.name
+        })
```

Comment:
You have to assign this new product list to `this.products`, otherwise it's not doing anything. Also, `newProduct` is not a correct name, it's not just one product, but all products except for one you're removing.